lib/controllers/share.js

```diff
@@ -103,7 +103,9 @@ module.exports = function(shareUrlPrefixes, newShareUrlPrefix) {
         makeGist(shareUrlPrefixes[newShareUrlPrefix], req.body).then(function(id) {
             id = newShareUrlPrefix + prefixSeparator + id;
             console.log('Created ID ' + id + ' using Gist service');
-            res.json({ id: id, path: req.baseUrl + '/' + id });
+            var resolveURL = req.baseUrl + '/' + id;
```

Comment:
Can't we get the name of the host from the request itself?  There's a `HOST` header I believe, as well as an `X-Forwarded-Host`.